docs/project/versioning.md

```diff
@@ -0,0 +1,269 @@
+        development build on a particular day during development of that
+        version.
+    -   `MAJOR.MINOR.PATCH-0.dev`: An interactive, incremental development build
+        on a particular day by some developer during development of that
+        version.
+
+See the sections below for the details of each aspect of these versions.
+
+## Major version increments
+
+Aligned with SemVer, the major version must increment for any _breaking change_
+to the language or any part of the toolchain.
+
+The first increment from 0 to 1 is expected to be based on achieving a desired
+milestone of feature completeness and quality to declare the language to have
+reached a stable version.
+
+Subsequent increments are expected to be done with a time-based release
+strategy. Features (or breaking changes) ready to ship will do so, and others
+will wait for the next major release. The exact cadence used is future work and
+should be determined based on discussions with customers in the ramp up to and
+after reaching the 1.0 milestone.
+
+However, just because we increment the major version for a major release and
+_can_ make breaking changes doesn't mean we _will_ or _should_. Breaking
+language changes are extraordinarily expensive for users due to the inherent
+scale of churn they impose. It is tempting to try to make non-breaking releases
+instead, but our experience with C++ language, compiler, and standard library
+updates is that truly making no breaking changes is extremely difficult and
+overly constraining. We expect most releases, especially in the early phases of
+the language to involve _some_ amount of breaking change and will simply work to
+make these as cheap to upgrade through and minimal as we can.
+
+At some future point, it is possible that Carbon will become so stable that it
+makes sense to consider using minor version increments for some releases. If and
+when this happens, we should revisit our versioning and release policies to
+establish a predictable and unsurprising structure.
+
+### Breaking changes
+
+Beyond traditional breaking API changes in either standard library APIs or tool
+APIs, Carbon also includes breaking changes in the language or toolchain.
+Language and toolchain breaking changes are any that cause correct, functioning,
+and non-reflective code to become invalid, rejected, incorrect, or silently
+change behavior.
+
+### Exclusions to what constitutes a breaking change
+
+Carbon excludes "reflective code" which is in some way detecting the Carbon
+version, or the presence or absence of features and as a consequence can be
+"broken" by detecting changes that are correctly designed to otherwise be
+non-breaking. We don't want adding features to be considered a breaking change
+and so exclude code that specifically detects such additions.
+
+Also excludes breaking changes to incorrect code unless it was accepted and
+functioning in some useful, and typically widespread, way despite its bugs.
```

Comment:
This sentence is missing a subject. "Our definition also excludes [...]", or "Carbon also excludes [...]" maybe?

Reply:
Done.